umh-core/pkg/config/manager.go

```diff
@@ -59,14 +84,16 @@ type FileConfigManager struct {
 	// logger is the logger for the config manager
 	logger *zap.SugaredLogger
 
-	// mutexReadAndWrite for full cycle read and write access to the config file
-	mutexReadAndWrite sync.Mutex
+	// mutexAtomicUpdate for full cycle read and write access (atomic update) to the config file
```

Comment:
when to use mutexAtomicUpdate: 
1. all writes need to go through an atomic update function, this requires that writeConfig is not exposed externally
2. therefore we can assume that all atomic update functions use this mutex

Reply:
the goal is:
1. to prevent two read/write cycles "atomic updates" happening at the same time